ui/components-react/src/components-react/properties/ValueRendererManager.tsx

```diff
@@ -171,6 +171,17 @@ export class PropertyValueRendererManager {
     return this._propertyRenderers.get(rendererType);
   }
 
+  /** Check whether a property record has a custom renderer registered that can render it */
+  public hasCustomRenderer(record: PropertyRecord): boolean {
+    const property = record.property;
+    const customRenderer =
```

Comment:
Looking at `selectRenderer` method it merged value renderer has higher priority than renderer based on `typename` Maybe registered custom renderer lookup could be shared between both methods? This part:

```typescript
if (
  record.property.renderer &&
  this._propertyRenderers.has(record.property.renderer.name)
)
  return this._propertyRenderers.get(record.property.renderer.name);
if (this._defaultMergedValueRenderer.canRender(record))
  return this._defaultMergedValueRenderer;
if (this._propertyRenderers.has(record.property.typename))
  return this._propertyRenderers.get(record.property.typename)!;
```